File: src/main/java/com/auth/service/util/CheckHash.java
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;

public class CheckHash {
    public static void main(String[] args) {
        BCryptPasswordEncoder encoder = new BCryptPasswordEncoder();
        String password = "admin123";
        String hash = "$2a$10$N.zmdr9k7uOCQb376NoUnuTJ8iAt6Z5EHsM8lE9lBOsl7iKTVKIvi";
        boolean isMatch = encoder.matches(password, hash);
        System.out.println("Match: " + isMatch);
    }
}

File: scratch/check_hash.py
import bcrypt

password = "admin123"
hash_str = "$2a$10$N.zmdr9k7uOCQb376NoUnuTJ8iAt6Z5EHsM8lE9lBOsl7iKTVKIvi"

is_match = bcrypt.checkpw(password.encode('utf-8'), hash_str.encode('utf-8'))
print(f"Match: {is_match}")
